feat: add LinkeDOMCrawler

Strajk opened the pull request:
The clean approach of https://github.com/apify/crawlee/pull/1879
as discussed https://github.com/apify/crawlee/pull/1890#issuecomment-1533271694

2023-05-09 16:05 Rebased

Strajk commented:
🏓

B4nan reviewed (commented):
we need some tests too :] maybe just duplicate this e2e test for jsdom? 

https://github.com/apify/crawlee/tree/master/test/e2e/jsdom-default-ts

Strajk commented:
* added e2e tests
* bumped version
* rebased

![tenor-7034491](https://github.com/apify/crawlee/assets/697301/24bddf8f-bc73-4ee2-9883-3e0717705f5e)

B4nan commented on test/e2e/linkedom-default-ts/actor/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+# using multistage build, as we need dev deps to build the TS source code
+FROM apify/actor-node:16-beta AS builder
```
```suggestion
FROM apify/actor-node:18-beta AS builder
```

B4nan commented on test/e2e/linkedom-default-ts/actor/Dockerfile:
```diff
@@ -0,0 +1,28 @@
+    && npm run build
+
+# create final image
+FROM apify/actor-node:16-beta
```
```suggestion
FROM apify/actor-node:18-beta
```

Strajk commented on test/e2e/linkedom-default-ts/actor/main.ts:
```diff
@@ -0,0 +1,27 @@
+    await Actor.init({ storage: new ApifyStorageLocal() });
+} else {
+    await Actor.init();
+}
```
🔥🔥🔥

B4nan commented:
sorry for the waiting, and thanks again :]

Strajk commented:
![tenor-263485356](https://github.com/apify/crawlee/assets/697301/38f8cbfe-d4bf-4786-9a2a-a8eceb174776)
![tenor-104075553](https://github.com/apify/crawlee/assets/697301/a36a3b26-ab67-4c6f-ab53-ec64eea55d4a)
![tenor-8894384](https://github.com/apify/crawlee/assets/697301/fe450679-96dd-47e6-9390-fbc722890000)

thxxx! no worries mr. banan :))
